packages/router/src/router.js

```diff
@@ -39,7 +40,9 @@ export function RouterProvider( { children } ) {
 
 	useEffect( () => {
 		return history.listen( ( { location: updatedLocation } ) => {
-			setLocation( getLocationWithParams( updatedLocation ) );
+			startTransition( () => {
```

Comment:
This is not needed. I was just playing with the idea and forgot to delete it 😅 .